src/main/java/uk/gov/pay/ledger/event/model/EventDto.java

```diff
@@ -0,0 +1,98 @@
+package uk.gov.pay.ledger.event.model;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategies;
+import com.fasterxml.jackson.databind.annotation.JsonNaming;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import io.swagger.v3.oas.annotations.media.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.service.payments.commons.api.json.ApiResponseDateTimeSerializer;
+
+import java.io.IOException;
+import java.time.ZonedDateTime;
+import java.util.Map;
+
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
+public class EventDto {
+
+    private static final Logger logger = LoggerFactory.getLogger(EventDto.class);
+    
+    private final String resourceExternalId;
+    @Schema(example = "AGREEMENT")
+    private final ResourceType resourceType;
+    @Schema(example = "CREATED")
+    private final String eventType;
+    @JsonSerialize(using = ApiResponseDateTimeSerializer.class)
+    @Schema(example = "\"2022-03-29T16:58:49.298Z\"")
+    private final ZonedDateTime timestamp;
+    private final Boolean live;
+    private final String serviceId;
+    private final Map<String, Object> data;
+
+    public EventDto(String resourceExternalId,
+                    ResourceType resourceType,
+                    String eventType,
+                    ZonedDateTime timestamp,
+                    Boolean live,
+                    String serviceId,
+                    Map<String, Object> data) {
+        this.resourceExternalId = resourceExternalId;
+        this.resourceType = resourceType;
+        this.eventType = eventType;
+        this.timestamp = timestamp;
+        this.live = live;
+        this.serviceId = serviceId;
+        this.data = data;
+    }
+
+    public static EventDto from(Event event, ObjectMapper objectMapper) {
+        try {
+            return new EventDto(
+                    event.getResourceExternalId(),
+                    event.getResourceType(),
+                    event.getEventType(),
+                    event.getEventDate(),
+                    event.getLive(),
+                    event.getServiceId(),
+                    objectMapper.readValue(event.getEventData(), new TypeReference<>() {
+                    })
+            );
+        }
+        catch (IOException e) {
```

Comment:
It would be good to have tests for this (if not exists already) as there are a few different paths here

Reply:
I've moved the return up into the catch, to make the flow clearer.  Do we still need tests?